src/matcher/matcher.go

```diff
@@ -0,0 +1,104 @@
+package matcher
+
+import (
+	"bufio"
+	"regexp"
+	"strings"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+)
+
+type matchResult int
+
+const (
+	noMatch matchResult = iota
+	exclude
+	include
+)
+
+//Matcher groups the rules to validate the service name
+type Matcher struct {
+	patterns []pattern
+}
+type pattern struct {
+	exclude bool
+	regex   string
+}
+
+// Match returns true if the string matches one of the patterns
+// within one level of precedence, the last matching pattern decides the outcome
+func (m *Matcher) Match(s string) bool {
+	n := len(m.patterns)
+	for i := n - 1; i >= 0; i-- {
+		if match := m.patterns[i].match(s); match > noMatch {
+			return match == include
+		}
+	}
+	return false
+}
+func (p pattern) match(s string) matchResult {
+	match, err := regexp.MatchString(p.regex, s)
+	if err != nil {
+		log.Warn(err.Error())
+		return noMatch
+	}
+
+	if p.exclude && match {
+		return exclude
+	}
+
+	if match {
+		return include
+	}
+	return noMatch
+}
+
+// New create a new Matcher instance from a multiline filter
+// - |
+//   windowsService.name:
+//   regex "^win.*$"
+//   "newrelic-infra"
+//   ! "winmgmt"
+func New(filterList string) Matcher {
+	var m Matcher
+	scanner := bufio.NewScanner(strings.NewReader(filterList))
+	r, _ := regexp.Compile("\"(.+)\"")
+
+	for scanner.Scan() {
+		var p pattern
+		isRegex := false
+		line := strings.TrimSpace(scanner.Text())
+
+		if strings.HasSuffix(line, ":") {
+			// "windowsService.name":
```

Comment:
Should we check that is present and it is the expected one anyway? 
Otherwise if user mistype it or forget about it, updating this in the future could break configurations

Reply:
you are right, I will make this check more strict to avoid future breaks , thanks!